xrpl/asyncio/transaction/reliable_submission.py

```diff
@@ -34,8 +35,18 @@ async def _wait_for_final_transaction_outcome(
     # new persisted transaction
 
     # query transaction by hash
-    transaction_response = await client._request_impl(Tx(transaction=transaction_hash))
-
+    try:
+        transaction_response = await client._request_impl(
+            Tx(transaction=transaction_hash)
+        )
+    except XRPLRequestFailureException as e:
+        if e.error == "txnNotFound" and attempts < 4:
+            # err code for if the txn is not found on the ledger due to race condition
```

Comment:
What is the race condition? (Is it the ledger hasn't closed yet? That this validator doesn't know about our transaction?)

If you don't know that's fine, but if you do I would recommend including that context in this comment.